### ISSUE TITLE
[Issue][<MazPhoneNumberInput>] Set default-country-value programmatically

### ISSUE THREAD

**Issue by afraguela** [LouisMazel/maz-ui#418]
**Describe the bug**
I am using a third-party IP lookup service to get the user's country. I make the request with Axios and then want to update the default-country-value. When I do this, it doesn't work. I've gone through the docs and there doesn't seem to be an API method I can use, unless I've missed it somewhere. Thanks in advance!

**Expected behavior**
Use the default-country-code as set in a ref, then update it when the Axios call is complete.

In the example below you can see that the REF is updated, but the country code in the Phone Input isn't.

**Code**

``` JS

const country_code = ref("FR")
function ip_lookup() {
      axios.get('http://ipwho.is')
          .then(function (response) {
              // handle success
              country_code.value = response.data.country_code
              console.log(country_code.value);
          })
          .catch(function (error) {
              // handle error
              console.log(error);
          })
}

onMounted(() => {
    ip_lookup()
})

<template>
 {{ country_code }}
<MazPhoneNumberInput v-model="phone" show-code-on-list color="info" @update="results = $event"
                :default-country-code="country_code" required size="md" />
</template>
```

**Comment by LouisMazel** [comment]
Hey @afraguela,

The component can already do that.
Use the property `fetch-country` 🔥

```html
<MazPhoneNumberInput 
  v-model="phone" 
  show-code-on-list 
  color="info" 
  :default-country-code="country_code" 
  required 
  size="md" 
  fetch-country <!-- 🔥 Add this property 🔥 -->
  @update="results = $event"
/>
```

Let me know if it's good for you !

**Comment by afraguela** [comment]
Thanks @LouisMazel !

I wanted to use another provider as the one supplier with the plug-in is blocked by some ad blockers. 

Would be good to have the option to set a custom provider.

**Comment by LouisMazel** [comment]
I just released a new version (v3.3.2) to fix request: 
- Now `default-country-code` is reactive, so the country will be updated  when it is changed
- `fetch-country` feature use [http://ipwho.is](http://ipwho.is) instead of [https://ip2c.org/s](https://ip2c.org/s)

So you can directly use `fetch-country`, but you want keep your code the country code changes will be apply

- [release v3.3.1](https://github.com/LouisMazel/maz-ui/releases/tag/v3.3.1)
- [Changelog](https://github.com/LouisMazel/maz-ui/blob/master/packages/lib/CHANGELOG.md)

**Comment by afraguela** [comment]
@LouisMazel thank you so much for the quick turnaround. It works as expected!
